libsolidity/interface/CompilerStack.cpp

```diff
@@ -509,22 +509,31 @@ string CompilerStack::applyRemapping(string const& _path, string const& _context
 	};
 
 	size_t longestPrefix = 0;
+	size_t longestContext = 0;
 	string longestPrefixTarget;
+
 	for (auto const& redir: m_remappings)
 	{
-		// Skip if we already have a closer match.
-		if (longestPrefix > 0 && redir.prefix.length() <= longestPrefix)
+		string contextFileString = sanitizePath(redir.context);
+		string prefixFileString = sanitizePath(redir.prefix);
+		// Skip if there is a prefix collision and the current context is closer
+		if (longestContext > 0 && contextFileString.length() < longestContext)
 			continue;
 		// Skip if redir.context is not a prefix of _context
-		if (!isPrefixOf(redir.context, _context))
+		if (!isPrefixOf(contextFileString, _context))
+			continue;
+		// Skip if we already have a closer match.
+		if (prefixFileString.length() < longestPrefix)
 			continue;
 		// Skip if the prefix does not match.
-		if (!isPrefixOf(redir.prefix, _path))
+		if (!isPrefixOf(prefixFileString, _path))
 			continue;
 
-		longestPrefix = redir.prefix.length();
+		longestContext = contextFileString.length();
+		longestPrefix = prefixFileString.length();
 		longestPrefixTarget = redir.target;
```

Comment:
I think it is better to name this `bestMatchTarget` now, because it corresponds to the longest prefix that is used for the longest context.